service/consul/datacenterWatch.go

```diff
@@ -1,109 +1,229 @@
-	if options.DatacenterWatchInterval == 0 {
-		return nil, errors.New("interval cannot be 0")
-	}
+func NewDatacenterWatcher(logger log.Logger, environment Environment, options Options, ctx context.Context) (*DatacenterWatcher, error) {
+	var consulWatch *consulDatacenterWatch
 
 	if logger == nil {
 		logger = defaultLogger
 	}
 
-	return &datacenterWatcher{
-		watchInterval: options.DatacenterWatchInterval,
-		logger:        logger,
-		shutdown:      make(chan struct{}),
-		environment:   environment,
-		options:       options,
-	}, nil
+	if options.DatacenterWatchInterval > 0 {
+		consulWatch = &consulDatacenterWatch{
+			watchInterval: options.DatacenterWatchInterval,
+			shutdown:      make(chan struct{}),
+		}
+	}
+
+	datacenterWatcher := &DatacenterWatcher{
+		consulDatacenterWatch: consulWatch,
+		logger:                logger,
+		options:               options,
+		environment:           environment,
+		inactiveDatacenters:   make(map[string]bool),
+	}
+
+	if options.ChrysomConfig != nil && options.ChrysomConfig.PullInterval > 0 {
+
+		if environment.Provider() == nil {
+			return nil, errors.New("must pass in a metrics provider")
+		}
+
+		options.ChrysomConfig.MetricsProvider = environment.Provider()
+		options.ChrysomConfig.Listener = datacenterWatcher.DatacentersListener()
+		options.ChrysomConfig.Logger = logger
+		chrysomClient, err := chrysom.CreateClient(*options.ChrysomConfig)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if ctx == nil {
+			ctx = context.Background()
+		}
+
+		datacenterWatcher.chrysomDatacenterWatch = &chrysomDatacenterWatch{
+			chrysomClient: chrysomClient,
+			ctx:           ctx,
+		}
+	} else if options.ChrysomConfig != nil {
+		return nil, errors.New("chrysom pull interval cannot be 0")
+	}
+
+	return datacenterWatcher, nil
 
 }
 
-func (i *datacenterWatcher) start() {
-	go i.watchDatacenters()
+func (d *DatacenterWatcher) StartConsulTicker() {
+	if d.consulDatacenterWatch != nil {
+		ticker := time.NewTicker(d.consulDatacenterWatch.watchInterval)
+		go d.watchDatacenters(ticker)
+	}
 }
 
-func (i *datacenterWatcher) stop() {
-	close(i.shutdown)
+func (d *DatacenterWatcher) StopConsulTicker() {
+	if d.consulDatacenterWatch != nil {
+		close(d.consulDatacenterWatch.shutdown)
+	}
 }
 
-func (i *datacenterWatcher) watchDatacenters() {
+func (d *DatacenterWatcher) StartChrysomTicker() {
+	if d.chrysomDatacenterWatch != nil {
+		d.chrysomDatacenterWatch.chrysomClient.Start(d.chrysomDatacenterWatch.ctx)
+	}
+}
+
+func (d *DatacenterWatcher) StopChrysomTicker() {
+	if d.chrysomDatacenterWatch != nil {
+		d.chrysomDatacenterWatch.chrysomClient.Stop(d.chrysomDatacenterWatch.ctx)
+	}
+}
 
-	environment := i.environment
-	client := i.environment.Client()
-	logger := i.logger
-	options := i.options
+func (d *DatacenterWatcher) watchDatacenters(ticker *time.Ticker) {
 
-	checkDatacenters := time.NewTicker(i.watchInterval)
+	client := d.environment.Client()
+	logger := d.logger
+	options := d.options
```

Comment:
```suggestion
```
I don't think we should create these. It doesn't gain us much and can make it harder to update the code later.